packages/malloy-vscode/src/extension/webviews/query_page/DownloadButton.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+import React, { useState } from "react";
+import styled from "styled-components";
+import { QueryDownloadOptions } from "../../webview_message_manager";
+import { Popover } from "../components/Popover";
+import { DownloadForm } from "./DownloadForm";
+import DownloadIcon from "../assets/download_hover.svg";
+
+interface DownloadButtonProps {
+  onDownload: (options: QueryDownloadOptions) => Promise<void>;
+}
+
+export const DownloadButton: React.FC<DownloadButtonProps> = ({
+  onDownload,
+}) => {
+  const [open, setOpen] = useState(false);
+
+  // console.log(Spinner);
+
+  return (
+    <>
+      <StyledDownloadIcon
+        onClick={() => setOpen(true)}
+        width={22}
+        height={22}
+      />
+      <Popover
+        open={open}
+        setOpen={setOpen}
+        width={200}
+        offsetSkidding={10}
+        offsetDistance={0}
+      >
+        <PopoverContent>
+          <DownloadForm
+            onDownload={async (options) => {
+              onDownload(options);
+              setOpen(false);
+            }}
+          />
+        </PopoverContent>
+      </Popover>
+    </>
+  );
+};
+
+const PopoverContent = styled.div`
+  padding: 15px;
+`;
+
+const StyledDownloadIcon = styled(DownloadIcon)`
+  cursor: pointer;
+  .hoverfill {
+    fill: transparent;
+  }
+  .primaryfill {
+    fill: #d4d6d8;
+  }
+  &:hover {
+    .hoverfill {
+      fill: rgb(240, 246, 255);
+    }
+    .primaryfill {
+      fill: #4285f4;
+    }
+  }
```

Comment:
Yep, agreed. These should definitely be swapped over when we're ready to commit to the theming. 

Seems like there are three categories of places that need to respect themes:
* UI elements like this
* Code
* Results rendering

For UI elements, the simple `var(--vscode-...)` should be fine. And for results, we just need to add some class names to the generated elements, and then we should be able to style them in the same way.

For code, the `shiki` library (used elsewhere in the repo) is capable of rendering according to VSCode themes. There were some slight problems integrating that inside the webview, which is why I didn't do it yet. I think shiki was having trouble loading either the themes or the grammars while inside the webview. We should be able to solve that either by adjusting how it's requesting those, or just moving the syntax highlighting into the extension instead of the webview. See here for the issue for this: https://github.com/looker-open-source/malloy/issues/352